0012-BIRS-distributed-scheduler.md

```diff
@@ -0,0 +1,611 @@
+	Day of week  | Yes        | 0-6 or SUN-SAT  | * / , - ?
+
+A user may use one of several pre-defined schedules in place of a cron expression.
+
+Entry                  | Description                                | Equivalent To
+-----                  | -----------                                | -------------
+@yearly (or @annually) | Run once a year, midnight, Jan. 1st        | 0 0 0 1 1 *
+@monthly               | Run once a month, midnight, first of month | 0 0 0 1 * *
+@weekly                | Run once a week, midnight on Sunday        | 0 0 0 * * 0
+@daily (or @midnight)  | Run once a day, midnight                   | 0 0 0 * * *
+@hourly                | Run once an hour, beginning of hour        | 0 0 * * * *
+
+Examples of how a user's `schedule` may look:
+```
+"0 30 * * * *"
+"0 * * 1,15 * Sun"
+"@hourly"
+"@every 1h30m"
+"@daily"
+```
+
+#### APIs
+
+*Note:* For cases where there are multiple instances of an application trying to write the same job name concurrently, we will follow the [last-write-wins concurrency pattern](https://docs.dapr.io/developing-applications/building-blocks/state-management/howto-stateful-service/?_gl=1*1bpqtb2*_ga*MTg0MDc4OTE4NS4xNjkzMjI0NDIw*_ga_60C6Q1ETC1*MTY5ODA3Mzc2OC4xMTkuMS4xNjk4MDc0NTk1LjAuMC4w#first-write-wins-and-last-write-wins), as used in our state-management Building Block and Actor Reminders.
+
+##### HTTP
+
+- Create a scheduled job
+    - POST
+    - http://localhost:{daprPort}/v1.0-alpha1/job/schedule/{name}
+
+- Delete a specific job by name
+    - DELETE 
+    - http://localhost:{daprPort}/v1.0-alpha1/job/{name}
+
+- Get a specific job by name
+    - GET
+    - http://localhost:{daprPort}/v1.0-alpha1/job/{name}
+
+- List all jobs for an application
+    - GET
+    - http://localhost:{daprPort}/v1.0-alpha1/jobs/{app_id}
+
+##### gRPC
+
+![Public Dapr APIs (User Facing)](./resources/0004-BIRS-distributed-scheduler/publicDaprAPI.png)
+
+###### User-Facing APIs
+
+```proto
+service Dapr {
+…
+// Create and schedule a job
+rpc ScheduleJobAlpha1(ScheduleJobRequest) returns (google.protobuf.Empty) {}
+
+// Gets a scheduled job
+rpc GetJobAlpha1(GetJobRequest) returns (GetJobResponse) {}
+
+// Delete a job
+rpc DeleteJobAlpha1(DeleteJobRequest) returns (google.protobuf.Empty) {}
+
+// List all jobs by app
+rpc ListJobsAlpha1(ListJobsRequest) returns (ListJobsResponse) {}
+}
+
+
+// Job is the definition of a job.
+message Job {
+  // The unique name for the job.
+  string name = 1;
+
+  // The schedule for the job.
+  optional string schedule = 2;
+
+  // Optional: jobs with fixed repeat counts (accounting for Actor Reminders).
+  optional uint32 repeats = 3;
+
+  // Optional: sets time at which or time interval before the callback is invoked for the first time.
+  optional string due_time = 4;
+
+  // Optional: Time To Live to allow for auto deletes (accounting for Actor Reminders).
+  optional string ttl = 5;
+
+  // Job data.
+  google.protobuf.Any data = 6;
```

Comment:
Just a question - noticed when working on the go-sdk implementation as well as @ItalyPaleAle highlighting it.
Is there a reason we're using a WKT here rather than byte string?

Reply:
Answered in the call. It allows data to be serialized as byte[] or as "builtin" protobuf.